pkg/p2p/libp2p/internal/handshake/handshake_test.go

```diff
@@ -382,15 +488,46 @@ func TestHandle(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		testInfo(t, nodeInfo, Info{
-			Address:   swarm.NewAddress(got.Syn.Address),
+		testInfo(t, node1Info, Info{
+			Overlay:   swarm.NewAddress(got.Syn.BzzAddress.Overlay),
+			Underlay:  got.Syn.BzzAddress.Underlay,
 			NetworkID: got.Syn.NetworkID,
 			Light:     got.Syn.Light,
 		})
 
-		_, err = handshakeService.Handle(stream1, info.ID)
-		if err != ErrHandshakeDuplicate {
-			t.Fatalf("expected %s err, got %s err", ErrHandshakeDuplicate, err)
+		_, err = handshakeService.Handle(stream1, node2AddrInfo.ID)
+		if err == nil || err.Error() != "handshake duplicate" {
```

Comment:
Why is ErrHandshakeDuplicate removed? It was better to check for a concrete error type then for an error string.

Reply:
I removed those specific errors because they were used only in tests to compare, and in libp2p to log a warning which, now when I look at it, does not seem necessary, so I thought maybe removing some code is nicer. Although I agree that specific error is a better approach in general. Basically, in libp2p we are logging those anyways and r not doing anything specific. WDYT?

Reply:
Yes, we are suing them in tests only, but even that is justifiable, I think.

Reply:
Ok, I'll switch back to a specific errors them